doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,229 @@
+.. _bin-blobs:
+
+.. warning::
+
+   This page documents Zephyr's support for binary blobs. The tooling and
+   infrastructure for this purpose are still incomplete, and so this page may
+   refer to future additions.
+
+Binary Blobs
+************
+
+In the context of an operating system that supports multiple architectures and
+many different IC families, some functionality may be unavailable without the
+help of executable code distributed in binary form.  Binary blobs (or blobs for
+short) are files containing proprietary machine code or data in a binary format,
+e.g. without corresponding source code released under an OSI approved license.
+
+Zephyr supports downloading and using third-party binary blobs via its built-in
+mechanisms, with some important caveats, described in the following sections. It
+is important to note that all the information in this section applies only to
+`upstream (vanilla) Zephyr <https://github.com/zephyrproject-rtos/zephyr>`_.
+
+There are no limitations whatsoever (except perhaps license compatibility) in
+the support for binary blobs in forks or third-party distributions of Zephyr. In
+fact, Zephyr’s build system supports arbitrary use cases related to blobs. This
+includes linking against libraries, flashing images to targets, etc. Users are
+therefore free to create Zephyr-based downstream software which uses binary
+blobs if they cannot meet the requirements described in this page.
+
+Software license
+================
+
+Most binary blobs are distributed under proprietary licenses which vary
+significantly in nature and conditions. It is up to the vendor to specify the
+license as part of the blob submission process. Blob vendors may impose a
+click-through or other EULA-like workflow when users fetch and install blobs.
+
+Hosting
+=======
+
+Blobs must be hosted on the Internet and managed by third-party infrastructure.
+Access to the files must be available without the need for user-level
+authentication.
+Two potential examples are Git repositories and web servers managed by
+individual hardware vendors.
+
+The Zephyr Project will not host binary blobs in its Git repositories or
+anywhere else.
+
+Fetching blobs
+==============
+
+Blobs are fetched from official third-party sources via a west extension command
```

Comment:
I see the warning at the top of the page, but I think this should be rewritten to say "will be fetched" for now, and then when the west extension is written, this document should be updated to link directly to the documentation for the extension. Similar changes elsewhere.

Non-blocking comment.

Reply:
Will do, this is a reasonable suggestion.

Reply:
I've done this for this particular section only, and for the west extension only. It did not make sense in the rest of the text IMHO.